enhancements/20231113-witness-node-support.md

```diff
@@ -0,0 +1,93 @@
+# Support witness node
+
+## Summary
+
+The witness node is a lightweight node that only runs the control plane components. The witness node is not schedulable and does not run any workload. The witness node forms a quorum with the other two nodes.
+
+### Related Issues
+
+https://github.com/harvester/harvester/issues/3266
+
+## Motivation
+
+Some users do not have three powerful nodes for the edge uses case. Instead, they would have two powerful nodes and one lightweight node. The lightweight node is only used for the `etcd` role. Kubernetes need at least three nodes to form a quorum. The witness node support will help the edge uses case. That will benefit both cost and high availability.
+
+### Goals
+
+Create a three-node cluster with one witness node.
+
+## Proposal
+
+### User Stories
+
+For the edge use case, the witness node support benefits both cost and high availability. The classical scenario would be two nodes and one witness node.
+
+### User Experience In Detail
+
+1. Set the node label `node-role.harvesterhci.io/etcd=true` to the node.
+2. The promoter will add the node to the cluster as `etcd` role.
+
+### API changes
+
+None
+
+## Design
+
+### Limitations
+
+- Each cluster only supports one witness node.
```

Comment:
Seems in installation stage, it is not possbile/easy to check if there has been one witness node, to prevent other `witness node` from joining,  e.g. mis-behaviour to set `witness node` to a normal  node. Then let's add warning in the document , that this limitation should be met by user manually.

Reply:
Sure, I will also add some warning on the installer.
Thanks for the reminder.

Reply:
After discussion, I thought we might also allow the user to choose the etcd nodes with join mode. (Means we do not limit only one etcd node)
Just satisfy the k8s quorum requirement, and we should let the user promote the etcd node to master node.